File: live_coding/palindrome.py
def palindrome(input):
    left = 0
    right = len(input) - 1
    is_palindrome = True
    while left < right:
        if input[left] != input[right]:
            is_palindrome = False
            break
        left += 1
        right -= 1

    if is_palindrome:
        print(f'{input} is palindrome')
    else:
        print(f'{input} is not palindrome')


def palindrome2(s):
    # Remove spaces and convert the string to lowercase
    # for case-insensitive comparison
    s = s.replace(" ", "").lower()
    if s == s[::-1]:
        print(f'{s} is palindrome')
    else:
        print(f'{s} is not palindrome')


palindrome('madam')
palindrome('madam1')

palindrome2('madam')
palindrome2('madam1')

File: live_coding/palindrome.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

void palindrome(const std::string& in) {
    bool is_palindrome = true;
    int left = 0;
    int right = in.size() - 1;
    for (; left < right; left++, right--) {
        if (in[left] != in[right]) {
            is_palindrome = false;
            break;
        }
    }
    
    if (is_palindrome) {
        cout << in << " is palindrome" << endl;
    } else {
        cout << in << " in not palindrome" << endl;
    }
}

void palindrome2(const std::string& in) {
    if (equal(in.begin(), in.end(), in.rbegin())) {
        cout << in << " is palindrome" << endl;
    } else {
        cout << in << " is not palindrome" << endl;
    }
}

int main()
{
    palindrome("madam");
    palindrome("madam2");

    palindrome2("madam");
    palindrome2("madam2");

    return 0;
}
